Mage/src/main/java/mage/game/GameState.java

```diff
@@ -839,6 +842,16 @@ public void addSimultaneousDamage(DamagedEvent damagedEvent, Game game) {
                     isPlayerBatchUsed = true;
                 }
             }
+
+            // per permanent
+            if (isPermanentDamage && event instanceof DamagedBatchForOnePermanentEvent) {
+                DamagedBatchForOnePermanentEvent oldPermanentBatch = (DamagedBatchForOnePermanentEvent) event;
+                if (oldPermanentBatch.getDamageClazz().isInstance(damagedEvent)
+                        && CardUtil.getEventTargets(event).contains(damagedEvent.getTargetId())) {
```

Comment:
hm, why `getEventTargets()` rather than just `getTargetId()` ?

Since it's for one permanent, I would expect it is possible to enforce that all events in the batch must have the same target id, and the batch event can also have that target id.

Reply:
Will that cause it to drop events that contain multiple damaged targets? Or is there a way to filter out the other targets in this conglomeration segment?

Reply:
Would that drop events that have multiple targets, some of which include the permanent in question? or is there a way to filter the targets out in this snippet?

Reply:
I'm confused, how can non-batch events have multiple targets?

Reply:
they wont, but the simultaneous events that this loop is going through might.

Reply:
If it's a `DamagedBatchForOnePermanent` (which here is checked by instanceof, so it must be), then it should only have one target right? isn't that the point, to only include events for that permanent?